src/index.js

```diff
@@ -1,39 +1,111 @@
+      </div>
+      <div class="main-edit">
+          <input type="checkbox" class="checkbox-edit" id="">
+          <input type="text" class="item-input-edit" value="${task.tasks[i].description}" id="${task.tasks[i].index}">
+          <img src="./delete.png" alt="" class="delete-item" id="${task.tasks[i].index}">
+      </div>
+  </div>`;
     }
   }
   listContainer.innerHTML = html;
   listContainer.classList.add('text-gray');
+  const allData = document.querySelectorAll('.item-check');
+  const descriptionAll = document.querySelectorAll('.descr');
+  for (let j = 0; j < allData.length; j += 1) {
+    allData[j].addEventListener('change', function () {
+      if (allData[j].checked) {
+        descriptionAll[j].classList.add('strike-through');
+        const btnId = parseInt(this.id, 10);
+        const myArray = task.tasks;
+        const myObject = myArray.find((obj) => obj.index === btnId);
+        myObject.completed = true;
+        const updateArray = JSON.stringify(myArray);
+        task.updateComplete(updateArray);
+      } else {
+        descriptionAll[j].classList.remove('strike-through');
+        const btnId = parseInt(this.id, 10);
+        const myArray = task.tasks;
+        const myObject = myArray.find((obj) => obj.index === btnId);
+        myObject.completed = false;
+        const updateArray = JSON.stringify(myArray);
+        task.updateComplete(updateArray);
+      }
+    });
+  }
+  const allAction = document.querySelectorAll('.dot');
+  const mainEdit = document.querySelectorAll('.main-edit');
+  const mainItem = document.querySelectorAll('.main-item');
+  allAction.forEach((ele, index) => {
+    ele.addEventListener('click', () => {
+      mainEdit[index].classList.add('active');
+      mainItem[index].classList.add('no-active');
+    });
+  });
+  const itemInputEdit = document.querySelectorAll('.item-input-edit');
+  itemInputEdit.forEach((ele) => {
+    ele.addEventListener('keypress', (event) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        task.editLocal(parseInt(event.target.id, 10), event.target.value);
+        updateData();
+      }
+    });
+  });
+  const deleteBtn = document.querySelectorAll('.delete-item');
+  deleteBtn.forEach((ele) => {
+    ele.addEventListener('click', (event) => {
+      const singleElementDelete = [];
+      singleElementDelete.push(parseInt(event.target.id, 10));
+      task.removeData(singleElementDelete);
+      updateData();
+    });
+  });
+};
+const clearBtn = document.querySelector('.clear');
+clearBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  const myNewArray = task.tasks;
+  const newIndexToRemove = [];
+  for (let k = 0; k < myNewArray.length; k += 1) {
+    if (myNewArray[k].completed === true) {
+      const indexRem = parseInt(myNewArray[k].index, 10);
+      newIndexToRemove.push(indexRem);
+    }
+  }
+  task.removeData(newIndexToRemove);
+  updateData();
+});
```

Comment:
- [ ] Please let's fix this bug :blush: As you can see from the video below, After I finished adding four tasks and mark two tasks as `completed`, I refreshed the page, and the two tasks I marked as `completed` were not showing on the UI but showing in the `localStorage` :smile: Please note that tasks that are marked as completed without clearing them should still show on the UI even if the page is refreshed :+1: 

[Screencast from 03-04-2023 02:00:19 AM.webm](https://user-images.githubusercontent.com/92303891/222870434-570fc95b-a948-4073-8b69-c00380e0914c.webm)

Reply:
- Please note that this issue persists, kindly look into it 😉.